modules/webbrowser/interaction/cefkeyboardmapping.cpp

```diff
@@ -0,0 +1,151 @@
+        case IvwKey::Enter:
+            return 13;
+        case IvwKey::Left:
+            return 37;
+        case IvwKey::Up:
+            return 38;
+        case IvwKey::Right:
+            return 39;
+        case IvwKey::Down:
+            return 40;
+        case IvwKey::Delete:
+            return 46;
+        case IvwKey::A:
+            return 97;
+        default:
+            return static_cast<int>(key);
+    }
+}
+
+unsigned int keyModifiers(KeyModifiers modifiers, IvwKey key) {
+    unsigned int cefModifiers = 0;
+    if (modifiers & KeyModifier::Control) {
+        cefModifiers |= EVENTFLAG_CONTROL_DOWN;
+    } else if (modifiers & KeyModifier::Shift) {
+        cefModifiers |= EVENTFLAG_SHIFT_DOWN;
+    } else if (modifiers & KeyModifier::Alt) {
+        cefModifiers |= EVENTFLAG_ALT_DOWN;
+    } else if (modifiers & KeyModifier::Super) {
+        cefModifiers |= EVENTFLAG_COMMAND_DOWN;
+    } else if (modifiers & KeyModifier::Menu) {
+        cefModifiers |= EVENTFLAG_ALT_DOWN;
+    } else if (modifiers & KeyModifier::Meta) {
+    }
+
+    if (key == IvwKey::CapsLock) {
+        cefModifiers |= EVENTFLAG_CAPS_LOCK_ON;
+    } else if (key == IvwKey::NumLock) {
+        cefModifiers |= EVENTFLAG_NUM_LOCK_ON;
+    }
+    switch (key) {
+        case IvwKey::Unknown:
+            break;
+        case IvwKey::Insert:
+        case IvwKey::Delete:
+        case IvwKey::Right:
+        case IvwKey::Left:
+        case IvwKey::Down:
+        case IvwKey::Up:
+        case IvwKey::PageUp:
+        case IvwKey::PageDown:
+        case IvwKey::Home:
+        case IvwKey::End:
+        case IvwKey::CapsLock:
+        case IvwKey::ScrollLock:
+        case IvwKey::NumLock:
+        case IvwKey::PrintScreen:
+        case IvwKey::Pause:
+            cefModifiers |= EVENTFLAG_IS_KEY_PAD;
+            break;
+        case IvwKey::KP0:
+        case IvwKey::KP1:
+        case IvwKey::KP2:
+        case IvwKey::KP3:
+        case IvwKey::KP4:
+        case IvwKey::KP5:
+        case IvwKey::KP6:
+        case IvwKey::KP7:
+        case IvwKey::KP8:
+        case IvwKey::KP9:
+        case IvwKey::KPDecimal:
+        case IvwKey::KPDivide:
+        case IvwKey::KPMultiply:
+        case IvwKey::KPSubtract:
+        case IvwKey::KPAdd:
+        case IvwKey::KPEnter:
+        case IvwKey::KPEqual:
+            cefModifiers |= EVENTFLAG_IS_KEY_PAD;
+            break;
+        case IvwKey::LeftShift:
+        case IvwKey::LeftControl:
+        case IvwKey::LeftAlt:
+        case IvwKey::LeftSuper:
+            cefModifiers |= EVENTFLAG_IS_LEFT;
+            break;
+        case IvwKey::RightShift:
+        case IvwKey::RightControl:
+        case IvwKey::RightAlt:
+        case IvwKey::RightSuper:
+            cefModifiers |= EVENTFLAG_IS_RIGHT;
+            break;
+        case IvwKey::Menu:
+            break;
+        case IvwKey::LeftMeta:
+        case IvwKey::RightMeta:
+            cefModifiers |= EVENTFLAG_IS_KEY_PAD;
```

Comment:
why keypad?

Reply:
I think that I followed this:
https://github.com/bkeiren/cef/blob/master/tests/cefclient/browser/util_win.cc